src/zenml/artifacts/external_artifact.py

```diff
@@ -84,70 +66,47 @@ def my_pipeline():
-            for field in ["value", "id", "name"]
-        ]
-        if sum(options) > 1:
-            raise ValueError(
-                "Only one of `value`, `id`, or `name` can be provided when "
-                "creating an external artifact."
-            )
-        elif sum(options) == 0:
-            raise ValueError(
-                "Either `value`, `id`, or `name` must be provided when "
-                "creating an external artifact."
-            )
-        return values
-
     def upload_by_value(self) -> UUID:
         """Uploads the artifact by value.
 
         Returns:
             The uploaded artifact ID.
-        """
-        from zenml.artifacts.utils import save_artifact
-
-        artifact_name = f"external_{uuid4()}"
-        uri = os.path.join("external_artifacts", artifact_name)
-        logger.info("Uploading external artifact to '%s'.", uri)
-
-        artifact = save_artifact(
-            name=artifact_name,
-            data=self.value,
-            extract_metadata=self.store_artifact_metadata,
-            include_visualizations=self.store_artifact_visualizations,
-            materializer=self.materializer,
-            uri=uri,
-            has_custom_name=False,
-            manual_save=False,
-        )
-
-        # To avoid duplicate uploads, switch to referencing the uploaded
-        # artifact by ID
-        self.id = artifact.id
-        self.value = None
-
-        logger.info("Finished uploading external artifact %s.", self.id)
-        return self.id
-
-    @property
-    def config(self) -> ExternalArtifactConfiguration:
-        """Returns the lightweight config without hard for JSON properties.
 
-        Returns:
-            The config object to be evaluated in runtime by step interface.
+        Raises:
+            RuntimeError: If the artifact value is not set.
         """
-        return ExternalArtifactConfiguration(
-            id=self.id,
-            name=self.name,
-            version=self.version,
-            model=self.model,
-        )
+        if self.value:
+            from zenml.artifacts.utils import save_artifact
+
+            artifact_name = f"external_{uuid4()}"
+            uri = os.path.join("external_artifacts", artifact_name)
+            logger.info("Uploading external artifact to '%s'.", uri)
+
+            artifact = save_artifact(
+                name=artifact_name,
+                data=self.value,
+                extract_metadata=self.store_artifact_metadata,
+                include_visualizations=self.store_artifact_visualizations,
+                materializer=self.materializer,
+                uri=uri,
+                has_custom_name=False,
+                manual_save=False,
+            )
+
+            # To avoid duplicate uploads, switch to referencing the uploaded
+            # artifact by ID
+            self._id = artifact.id
+            self.value = None
+
+            logger.info("Finished uploading external artifact %s.", self._id)
```

Comment:
Can we standardize the use of f-Strings
```suggestion
            logger.info(f"Finished uploading external artifact {self._id}.")
```